es-app/src/CollectionSystemManager.cpp

```diff
@@ -614,139 +570,140 @@ void CollectionSystemManager::updateCollectionFolderMetadata(SystemData* sys)
+{
+	CollectionFileData* newGame = new CollectionFileData(game, this->system);
+	this->system->getRootFolder()->addChild(newGame);
+	FileFilterIndex* index = this->system->getIndex();
+	index->addToIndex(newGame);
+	SystemData* systemViewToUpdate = CollectionSystemManager::get()->getSystemToView(this->system);
+
+	// add to bundle index as well, if needed
+	if (systemViewToUpdate != this->system)
 	{
-		LOG(LogInfo) << "Couldn't find custom collection config file at " << path;
-		return;
+		systemViewToUpdate->getIndex()->addToIndex(newGame);
 	}
-	LOG(LogInfo) << "Loading custom collection config file at " << path;
+	ViewController::get()->onFileChanged(game, FILE_METADATA_CHANGED);
+	ViewController::get()->getGameListView(systemViewToUpdate)->onFileChanged(newGame, FILE_METADATA_CHANGED);
+}
 
-	FileData* rootFolder = newSys->getRootFolder();
-	FileFilterIndex* index = newSys->getIndex();
+bool CollectionSystemManager::CollectionSystem::isValidGame(FileData* game)
+{
+	bool valid = game->isGameFile();
+	switch (this->decl.type) {
+	case AUTO_LAST_PLAYED:
+		valid = valid && game->metadata.getInt("playcount") > 0;
+		break;
+	case AUTO_FAVORITES:
+		// we may still want to add non-games to "favorites"
+		valid = game->metadata.getBool("favorite");
+		break;
+	case AUTO_HIDDEN:
+		valid = game->metadata.getBool("hidden"); // also include non-games
+		break;
+	case AUTO_KIDGAME:
+		valid = game->metadata.getBool("kidgame");
+		break;
+
+	}
+	return valid;
+}
+
+// populates a Custom Collection System
+void CollectionSystemManager::CustomCollectionSystem::populate()
+{
+	
+	std::string path = getCustomCollectionConfigPath(this->system->getName());
+	if (path == "")
+		return; // path not found
+
+	FileData* rootFolder = this->system->getRootFolder();
+	FileFilterIndex* index = this->system->getIndex();
 
 	// get Configuration for this Custom System
 	std::ifstream input(path);
 
-	// get all files map
-	std::unordered_map<std::string,FileData*> allFilesMap = getAllGamesCollection()->getRootFolder()->getChildrenByFilename();
+	// get map of all files
+	std::unordered_map<std::string,FileData*> allFilesMap = CollectionSystemManager::get()->getAllGamesCollection()->getRootFolder()->getChildrenByFilename();
 
 	// iterate list of files in config file
-
 	for(std::string gameKey; getline(input, gameKey); )
 	{
-		std::unordered_map<std::string,FileData*>::iterator it = allFilesMap.find(gameKey);
-		if (it != allFilesMap.end()) {
-			CollectionFileData* newGame = new CollectionFileData(it->second, newSys);
-			rootFolder->addChild(newGame);
-			index->addToIndex(newGame);
+		auto game = allFilesMap.find(gameKey);
+		if (game != allFilesMap.end())
+		{
+			addGameToCollection(game->second);
 		}
 		else
 		{
-			LOG(LogInfo) << "Couldn't find game referenced at '" << gameKey << "' for system config '" << path << "'";
+			LOG(LogDebug) << "Couldn't find game referenced at '" << gameKey << "' for system config '" << path << "'";
```

Comment:
Small nit: I think LogDebug is only triggered when you call ES with a specific command line option, so this message will not really be helpful for users to understand why something is not behaving as expected.

LogError is too scary, as it actually shows on the command line, but I'll defer to you.

Reply:
hmm, true, I've not really considered running ES in non-debug mode in a long, long time :-) LogInfo is fine!